### PR TITLE
Sprint 25

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 25

### CODE REVIEW

**Comment by waldoj** [comment]
Here's the review checklist that I'll work my way through in reviewing this PR:

- [ ] begin by deleting any existing copy of the Git repository and cloning a fresh copy, running `docker system prune` to remove any cached artifacts from any prior code review, and running `./reset-dependencies.sh` to update Node packages
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] review static code analysis results in SonarCloud
- [ ] run a security audit of dependencies (`npm audit`) to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](https://github.com/ustaxcourt/ef-cms/blob/staging/docs/CODE_REVIEW.md#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] methodically review all new tests for correctness, quality of naming
	- [ ] look at code coverage of tests in SonarCloud
	- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by waldoj** [comment]
Because GitHub lacks effective pull-request-level feedback mechanisms that must be addressed before a pull request can be merged, I'll be using `README.md` as the place to attach PR-level comments.

File: README.md [hunk 1]
```diff
@@ -151,7 +151,7 @@ petitionsclerk1@example.com – petitionsclerk5@example.com
 docketclerk1@example.com – docketclerk5@example.com
 respondent1@example.com – respondent10@example.com
 practitioner1@example.com – practitioner10@example.com
-seniorattorney1@example.com – seniorattorney5@example.com
+adc1@example.com – adc5@example.com
```
Comment (waldoj) [inline README.md]: `test-all.sh` routinely fails for me — it dies with this:

```
 91  278M   91  254M    0     0  1720k      0  0:02:45  0:02:31  0:00:14 1219kWaiting for http://localhost:9200/ to be hosted...

 91  278M   91  255M    0     0  1717k      0  0:02:45  0:02:32  0:00:13 1200kWaiting for http://localhost:3000/api/swagger to be hosted...

 92  278M   92  257M    0     0  1716k      0  0:02:45  0:02:33  0:00:12 1264kWaiting for http://localhost:9200/ to be hosted...

 92  278M   92  258M    0     0  1714k      0  0:02:46  0:02:34  0:00:12 1296kWaiting for http://localhost:3000/api/swagger to be hosted...

 93  278M   93  259M    0     0  1713k      0  0:02:46  0:02:35  0:00:11 1391kWaiting for http://localhost:9200/ to be hosted...

 94  278M   94  261M    0     0  1713k      0  0:02:46  0:02:36  0:00:10 1513kWaiting for http://localhost:3000/api/swagger to be hosted...

 94  278M   94  263M    0     0  1713k      0  0:02:46  0:02:37  0:00:09 1567kWaiting for http://localhost:9200/ to be hosted...

 94  278M   94  264M    0     0  1708k      0  0:02:46  0:02:38  0:00:08 1449kWaiting for http://localhost:3000/api/swagger to be hosted...

:(  giving up.
```

Please resolve this.

File: package.json [hunk 2]
```diff
@@ -107,23 +110,24 @@
     "serverless-offline": "^5.11.0",
     "serverless-offline-dynamodb-streams": "^3.0.0",
     "serverless-plugin-aws-alerts": "^1.3.1",
-    "serverless-plugin-bind-deployment-id": "github:ustaxcourt/serverless-plugin-bind-deployment-id#update_lodash",
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply package.json]: Yes!

File: README.md [hunk 3]
```diff
@@ -151,7 +151,7 @@ petitionsclerk1@example.com – petitionsclerk5@example.com
 docketclerk1@example.com – docketclerk5@example.com
 respondent1@example.com – respondent10@example.com
 practitioner1@example.com – practitioner10@example.com
-seniorattorney1@example.com – seniorattorney5@example.com
```
[the remark this answers is not in the document]
Comment (sutt0n) [reply README.md]: These are still accurate!

File: web-client/src/presenter/actions/CaseConfirmation/generateCaseConfirmationPdfUrlAction.test.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        caseDetail: {
+          caseId: 'ca123',
+          docketNumber: '123-45',
+        },
+        token: 'abcdefg',
+      },
+    });
+
+    const expectedUrl =
+      'http://www.demo.com/documents/case-123-45-confirmation.pdf/document-download-url?token=abcdefg';
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply web-client/src/presenter/actions/CaseConfirmation/generateCaseConfirmationPdfUrlAction.test.js]: Updated!

File: README.md [hunk 5]
```diff
@@ -131,7 +131,7 @@ respondent
 respondent1 - respondent4
 practitioner
 practitioner1 - practitioner4
-seniorattorney
+adc
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply README.md]: Updated!

File: README.md (hunk 1 above)
Comment (ericsorenson) [reply README.md]: We've added 2 additional cache resets to this script. Joe and I are both able to run this script successfully.

Comment (waldoj) [reply README.md]: That did it! Thank you. :)

**Comment by waldoj** [comment]
@JessicaMarine, I have reviewed the sprint 25 pull request, and I believe that it meets all standards of our code review process and that it adheres to the QASP. I recommend that the Court merge it to `staging`.
